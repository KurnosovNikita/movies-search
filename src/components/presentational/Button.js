import React from "react";

const MenuButton = ({onClick, className, text, id}) => (
    <button onClick={onClick} className={className} id={id}>{text}</button>
);

export default MenuButton;