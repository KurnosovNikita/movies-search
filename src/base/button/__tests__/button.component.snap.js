import React from 'react';
import {shallow} from 'enzyme';
import Button from '../button.component';

describe('Button component', () => {
    let props;
    beforeEach(() => {
        props = {
            onClick: jest.fn(),
            className:'className',
            text: 'text',
            id: 'id',
        };
    });

    it('should render <Button> correctly', () => {
        const component = shallow(<Button {...props}/>);

        expect(component).toMatchSnapshot();
    });

    describe('render props correctly', () => {

        it('render with className', () => {
            const component = shallow(<Button {...props}/>);

            expect(component.props().className).toEqual('className');
        });

        it('render with id', () => {
            const component = shallow(<Button {...props}/>);

            expect(component.props().id).toEqual('id');
        });
    });
});
