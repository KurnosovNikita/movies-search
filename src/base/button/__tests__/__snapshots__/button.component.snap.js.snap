// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render <Button> correctly 1`] = `
<button
  className="className"
  id="id"
  onClick={[MockFunction]}
>
  text
</button>
`;
